### PR TITLE
Update deploy-staging.yaml to run after test completion

### PR DESCRIPTION
**Pull request by spokenbird**

### CODE REVIEW

File: .github/workflows/deploy-staging.yaml [hunk 1]
```diff
@@ -1,8 +1,10 @@
 name: Deploy to Aptible Staging
 
 on:
-  push:
-    branches: [ main ]
+  workflow_run: 
+    workflows: [ "Run tests" ]
+    types: [ completed ]
+    branches: [ update-deploy-to-staging ]
```
Comment (bseeger) [inline .github/workflows/deploy-staging.yaml:7]: what is this branch? Is it a temporary branch created in this process?

Comment (spokenbird) [reply .github/workflows/deploy-staging.yaml:7]: Yep, for testing purposes

Comment (bseeger) [reply .github/workflows/deploy-staging.yaml:7]: Oh, you're using this for testing.  Is that right?

Comment (bseeger) [reply .github/workflows/deploy-staging.yaml:7]: So for for the final PR, it will say "main"?

Comment (spokenbird) [reply .github/workflows/deploy-staging.yaml:7]: It should yes, but I'm realizing this isn't going to work the way I had hoped and now I'm not sure how to test this without merging it to main first 😓

Comment (bseeger) [reply .github/workflows/deploy-staging.yaml]: Might be fine to merge to main first -- you won't be hurting anything with iterating on this there.  I think just warn us all first so we know that things might fail for a limited amount of time.

**Pull request merged**